profab/ebs.py

```diff
@@ -22,14 +23,44 @@ def create(cls, server, size):
         """
         _logger.info("Creating volume size %dGB on %s",
             size, server.instance.placement)
-        volumes = server.get_volumes()
+        device = cls.get_volume(server.get_volumes())
+        volume = cls(server, server.cnx.create_volume(
+            size, server.instance.placement), device)
+        cls.attach_volume(server.insatance.id, volume, device)
+        return volume
+
+
+    @classmethod
+    def creat_from_snapshot(cls, server, snapshot):
+        """Create a new volume from snapshot.
+        """
+        _logger.info("Creating volume size 200GB from snapshot id %s",
```

Comment:
I had to do this because I don't know how to create volume from snapshot to
set size of volume. Can you explain me about this.

On Fri, Aug 28, 2015 at 9:26 AM, Kirit Sælensminde <notifications@github.com
> wrote:

> In profab/ebs.py
> <https://github.com/Proteus-tech/profab/pull/11#discussion_r38166847>:
>
> > @@ -22,14 +23,44 @@ def create(cls, server, size):
> >          """
> >          _logger.info("Creating volume size %dGB on %s",
> >              size, server.instance.placement)
> > -        volumes = server.get_volumes()
> > +        device = cls.get_volume(server.get_volumes())
> > +        volume = cls(server, server.cnx.create_volume(
> > +            size, server.instance.placement), device)
> > +        cls.attach_volume(server.insatance.id, volume, device)
> > +        return volume
> > +
> > +
> > +    @classmethod
> > +    def creat_from_snapshot(cls, server, snapshot):
> > +        """Create a new volume from snapshot.
> > +        """
> > +        _logger.info("Creating volume size 200GB from snapshot id %s",
>
> The message says that this is always a 200GB volume -- surely it is
> whatever size the snapshot is? I think we should either leave out the size
> from the log message, or work out what the correct size is to put in here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Proteus-tech/profab/pull/11/files#r38166847>.
>